R/create_single_dose_dataset.R

```diff
@@ -187,8 +187,7 @@ dose_freq_lookup <- tribble(
 #'
 #' @param dataset Input dataset
 #'
-#'   The columns specified by `dose_freq`, `start_date` and the `end_date`
-#'   parameters are expected.
+#'   `r roxygen_param_dataset(expected_vars_args = c("dose_freq", "start_date", "end_date"))`
```

Comment:
@bundfussr Is this what you had in mind? Easier to see in action in the `.Rd` (screenshotted here) 

![image](https://github.com/pharmaverse/admiral/assets/43018762/0f37b050-2220-4f13-9100-5aeffb22a490)

Reply:
> Is this what you had in mind?

Yes.

I would create the label ("Input dataset") by the function as well. Then it is consistent.

Maybe we could even create `@param dataset` by the function. But I am not sure if this works.